server/routes/api/users.js

```diff
@@ -75,4 +76,78 @@ router.post("/login", (req, res) => {
   }
 });
 
+// @route GET /api/users/{id}
+// @desc Get a User object
+// @access Authenticated User can get own record
+router.get("/:id", (req, res) => {
+  let id = req.params.id;
+  User.findById(id, function(err, user) {
+    if (!user) {
+      res.status(404).send({ id: `User with id ${id} is not found` });
+    } else {
+      res.json(user);
+    }
+  });
+});
+
+// @route PUT /api/users/{id}
+// @desc Update a User object
+// @access Authenticated Users
+router.put("/:id", (req, res) => {
+  let id = req.params.id;
+
+  // TO DO *****************
+  const { errors, isValid } = validateUserInput(req.body);
+
+  if (!isValid) {
+    return res.status(400).json(errors);
+  }
+
+  User.findById(id, function(err, user) {
+    if (!user) {
+      res.status(404).send({ id: `User with id ${id} is not found` });
+    } else {
+      //if updating email field, check if User with new email alredy exists
+      if (req.body.email !== user.email) {
+        User.findOne({ email: req.body.email }).then(userEmailCheck => {
+          if (userEmailCheck) {
```

Comment:
Given that this is asynchronous, we are going to have problems, since you may actually save a user right after this check with the new email address. 

You either need to put the logic into this code block or use async/await. We can talk about that in standup

We'd probably want to delete the gmail token too if someone changes email :)

Reply:
Good catch! Fixed both points. 
@airensurzyn I copied this code from the prospects.js routes so I fixed the async problem there too.